### PR TITLE
135177141 user role creation

### PR DESCRIPTION
**Pull request by rdavid1099**
Users can now have different roles.

A user's role determines what tools are displayed in the navbar.

### CODE REVIEW

**Comment by rdavid1099** [comment]
Not ready to pull... Let me fix mocking real quickly.

**Comment by rdavid1099** [comment]
Okay... It's now ready to be looked at.

All tests are passing except the two tests which have been failing since receiving the code base.

**Comment by rdavid1099** [comment]
I'm fixing the merge conflict after pulling that last request

File: app/views/layouts/_navbar.html.erb [hunk 1]
```diff
@@ -29,7 +29,22 @@
 
       <ul class="nav navbar-nav navbar-right">
           <p class="navbar-text">Signed in as <strong><%= current_user.username %></strong></p>
-
+          <% if current_user.admin? %>
+            <li class="dropdown">
+              <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Admin Tools<span class="caret"></span></a>
+              <ul class="dropdown-menu">
+                <li><%= link_to "Approve Recent Company Submissions", companies_path %></li>
+                <li><%= link_to "Update Existing Company Information", cities_path %></li>
```
Comment (rvna) [inline app/views/layouts/_navbar.html.erb:37]: It doesn't quite make sense that "Approve Recent Company Submissions" would go to the companies path. Is this a placeholder?

File: app/views/layouts/_navbar.html.erb [hunk 2]
```diff
@@ -29,7 +29,22 @@
 
       <ul class="nav navbar-nav navbar-right">
           <p class="navbar-text">Signed in as <strong><%= current_user.username %></strong></p>
-
+          <% if current_user.admin? %>
+            <li class="dropdown">
+              <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Admin Tools<span class="caret"></span></a>
```
Comment (rvna) [inline app/views/layouts/_navbar.html.erb:34]: I really like the dropdown for role-specific tools.

File: spec/features/standard_user_logs_in_spec.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+require 'rails_helper'
+
+RSpec.describe 'standard user logs in' do
+  it 'does not see any additional tools in navbar' do
+    user_logs_in
+    visit root_path
+
+    expect(page).to_not have_content('Admin Tools')
```
Comment (rvna) [inline spec/features/standard_user_logs_in_spec.rb:8]: This is a clean way to test this; nice.

File: spec/rails_helper.rb [hunk 4]
```diff
@@ -23,51 +20,34 @@
...
-  # https://relishapp.com/rspec/rspec-rails/docs
   config.infer_spec_type_from_file_location!
-
-  # Filter lines from Rails gems in backtraces.
   config.filter_rails_from_backtrace!
-  # arbitrary gems may also be filtered via:
-  # config.filter_gems_from_backtrace("gem name")
+end
+
+def admin_logs_in
```
Comment (rvna) [inline spec/rails_helper.rb:34]: Having a helper for each type of user is nice.

File: app/views/layouts/_navbar.html.erb (hunk 1 above)
Comment (rdavid1099) [reply app/views/layouts/_navbar.html.erb:37]: Yeah... All these design decisions are just placeholders... None of the routes are built.

More details are on the card in the pivotal tracker

**Comment by rdavid1099** [comment]
Basic roles are created
- Added roles to Users
- Defaults to 'standard'
- Enum used to change user role from standard, moderator or admin
- Moderators and admins see additional tools in their navbar

Still need to be completed
- Scope accessibility for the three roles
- All functionality for admin and moderators not built out... just dummy tabs put in navbar

**Pull request merged**